[DX-1297, TT-11697] Add documentation on how monitoring works in Tyk Cloud

komuw opened the pull request:
## **User description**
### For internal users - Please add a Jira DX PR ticket to the subject!
- https://tyktech.atlassian.net/browse/DX-1297
- https://tyktech.atlassian.net/browse/TT-11697

### Preview Link
<!-- Add a direct preview link to make it easier for the reviewer to review. Best to add a few direct links to the pages in the PR -->
- https://deploy-preview-4518--tyk-docs.netlify.app/docs/nightly/tyk-cloud/environments--deployments/monitoring-usage/
- https://deploy-preview-4518--tyk-docs.netlify.app/docs/nightly/tyk-cloud/environments--deployments/monitoring-how-it-works/


### Description
<!-- 1. Describe your changes in detail. Why is this change required? What problem does it solve?
     2. Please explain the change to the reviewer. Pay special attention to changes that are hard to spot, 
       for example:
       2.1. Name change in the file name or directory name - please flag it out since it’s hard 
            to compare text after such a change 
       2.2. Terminology change - flag it out to make sure the reviewer is aware before approving
     3. @mentions of the person to review the proposed changes. They need to be able to know the topic well in order to approve it.
-->
- Add documentation on how monitoring works in Tyk Cloud
  There have been a number of queries on how monitoring works, this documentation update tries to address that.
  I have tried to give a high level view of how monitoring works without going too much into implementation details(since the implementation might change in the future). There is an [internal documentation](https://tyktech.atlassian.net/wiki/spaces/EN/pages/2400878648/Tyk+cloud+monitoring) on how monitoring works that goes into the implementation detail.

<br>

#### Screenshots (if appropriate)
<br>

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply -->

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR
<!-- Label your PR according to the type of changes that your code introduces. This ensures that we know how/when to publish the PR. These are the options:
- Fixing typo (please merge to production) - add the label `now`
- Documenting a new feature (please merge to production) - add the label `now`
- Documentation for future release (please do not merge to production) - add label `future release` and label of the release
- [Something else (please add if needs merging to production or not) - add label according to the above logic -->


___

## **Type**
documentation, enhancement


___

## **Description**
- Added new detailed documentation on how monitoring works in Tyk Cloud, including specifics on traffic types and storage metrics.
- Introduced a new guide on how to access and interpret monitoring data in Tyk Cloud.
- Updated the site menu to reflect the new structure of the monitoring documentation, enhancing navigation and user experience.
- Removed outdated content from the old monitoring overview page to avoid redundancy and confusion.


___



## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>monitoring-how-it-works.md</strong><dd><code>Add Detailed Monitoring Mechanics Documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-how-it-works.md
<li>Introduced a new documentation page detailing how monitoring works in <br>Tyk Cloud.<br> <li> Explained the calculation of throughput metrics and the distinction <br>between charged and uncharged traffic.<br> <li> Included an image to visually represent traffic pricing in monitoring.<br> <br>


</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4518/files#diff-efaa7ee1572e085b039ac1a54036c426d947000476e03ddf33900740abbdb6c2">+38/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>monitoring-usage.md</strong><dd><code>New Guide on Monitoring Usage in Tyk Cloud</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-usage.md
<li>Created a new page to guide users on how to view throughput and <br>storage usage in Tyk Cloud.<br> <li> Provided instructions on how to access and interpret monitoring data.<br> <li> Added images to illustrate the monitoring interface and data filtering <br>options.<br>


</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4518/files#diff-76d327612711fd0e977d8471fbba283f5fc9ea8bb4aa03d1d2040f690611a5c9">+35/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>monitoring.md</strong><dd><code>Remove Old Monitoring Overview Content</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring.md
- Removed the original monitoring page content.



</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4518/files#diff-25bf4aa967368d03384792484323a495340354a1bdb08729a06bf6417fc597dd">+0/-17</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Configuration changes
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Update Menu Structure for Monitoring Documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/data/menu.yaml
<li>Updated the menu structure to include new pages under the monitoring <br>section.<br> <li> Changed the 'Monitoring' menu entry from a page to a directory.<br>


</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4518/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+10/-2</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

komuw commented on tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-how-it-works.md:
```diff
@@ -0,0 +1,38 @@
+### Throughput metric.
+The way that it is calculated is that Tyk Cloud keeps a counter of the total request and response sizes for traffic transferred through a deployment. We then calculate the difference between the throughput usage at the current time less the throughput at last midnight. That way, the throughput metrics that is displayed is the throughput for the current day.
+
+Not all traffic is included in throughput calculation. Only external traffic is charged, whereas internal traffic is not. Monitoring service sum-sup traffic between different services:
```
Is it better to say `Only external traffic is charged` or `Only external traffic is priced`?

komuw commented:
The docs bible; https://tyktech.atlassian.net/wiki/spaces/EN/pages/1888976900/The+Docs+Bible+-+Docs+Contributors+Guide#Label-PR says;
```
For future releases, label it with -

- Future-release
- {release vX.Y.Z} - intended version number, e.g. release-6.1.
```

I have already added the `future-release` label. 
I have also added a new label `Ara 1.18.0` since we would like this to be released when Tyk Cloud v1.18.0 is released. Let me know if that's the right way, or if you already have pre-existing labels for Tyk Cloud.

dcs3spp reviewed (changes requested):
Many thanks @komuw for submitting this much needed content!! I like the billing diagram overview and structure.

I have made some minor suggestions for section headings and suggest moving the billing diagram so that it appears before the billed and unbilled traffic example list

komuw commented on tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-how-it-works.md:
```diff
@@ -0,0 +1,38 @@
+Not all traffic is included in throughput calculation. Only external traffic is charged, whereas internal traffic is not. Monitoring service sum-sup traffic between different services:
+
+Charged Traffic:
+ - Traffic between User → Control Plane
```
fixed in https://github.com/TykTechnologies/tyk-docs/pull/4518/commits/7fe9433e5d58e50b5969e7c44067cef7418990ce

komuw commented on tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-how-it-works.md:
```diff
@@ -0,0 +1,38 @@
+Tyk Cloud keeps a counter of the total request/response sizes for traffic transferred through a deployment. We then calculate the difference between the throughput usage at the current time less the throughput at last midnight. That way, the throughput metrics that is displayed is the throughput for the current day.
+
+Not all traffic is included in throughput calculation. Only external traffic is charged, whereas internal traffic is not. Monitoring service sum-sup traffic between different services:
+
```
fixed in https://github.com/TykTechnologies/tyk-docs/pull/4518/commits/213289023924b3e3a43969cffaf3eebe13891423

komuw commented on tyk-docs/data/menu.yaml:
```diff
@@ -306,9 +306,17 @@ menu:
+              path: /tyk-cloud/environments--deployments/monitoring-usage
+              category: Page
+              show: True
+            - title: "How it works"
```
fixed in https://github.com/TykTechnologies/tyk-docs/pull/4518/commits/2d248b7c0be85fe73b67719d4db590008b13d481

komuw commented on tyk-docs/data/menu.yaml:
```diff
@@ -306,9 +306,17 @@ menu:
+            category: Directory
             show: True
+            menu:
+            - title: "Usage"
```
fixed in https://github.com/TykTechnologies/tyk-docs/pull/4518/commits/09b63d32eab5d476128fb34678decfa64d82287d

komuw commented on tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-usage.md:
```diff
@@ -0,0 +1,35 @@
+---
+title: "Usage"
```
fixed in https://github.com/TykTechnologies/tyk-docs/pull/4518/commits/09b63d32eab5d476128fb34678decfa64d82287d

komuw commented on tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-how-it-works.md:
```diff
@@ -0,0 +1,38 @@
+  - /tyk-cloud/environments-deployments/monitoring/
+  - /tyk-cloud/environments-&-deployments/monitoring
+---
+
```
fixed in https://github.com/TykTechnologies/tyk-docs/pull/4518/commits/f6535b02735211f6add25f047c4f6090fd4a7fff

komuw commented on tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-how-it-works.md:
```diff
@@ -0,0 +1,38 @@
+
+{{< img src="/img/cloud/tyk-cloud-monitoring-priced-traffic.png" alt="Monitoring Traffic Pricing" >}}
+
+### Storage metric.
```
fixed in https://github.com/TykTechnologies/tyk-docs/pull/4518/commits/0b580b7600168bcde18512035198f5bc762e3a2e

komuw commented on tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-how-it-works.md:
```diff
@@ -0,0 +1,38 @@
+{{< img src="/img/cloud/tyk-cloud-monitoring-priced-traffic.png" alt="Monitoring Traffic Pricing" >}}
+
+### Storage metric.
+When a client makes a request to a Tyk Gateway deployment, the details of the request and response are captured and [stored in Redis]({{< ref "tyk-dashboard-analytics/" >}}). Tyk Pump processes the records from Redis and forwards them to MongoDB. Finally, Tyk Cloud reads that data from MongoDB and displays it in the _Storage_ section of _Monitoring_. 
```
Mongo is the only option so far. It might change in the future. Maybe I shouldn't mention Mongo at all?

dcs3spp replied:
Ok thanks @komuw I think we can leave this unchanged and I will resolve the comment

komuw commented:
> Many thanks @komuw for submitting this much needed content!! I like the billing diagram overview and structure.
> 
> I have made some minor suggestions for section headings and suggest moving the billing diagram so that it appears before the billed and unbilled traffic example list

Thanks for the review! I have addressed your feedback in subsequent commits.

dcs3spp commented:
Thanks @komuw I have approved the PR and will merge and release once approved by @mitjaziv and @imogenkraak

dcs3spp commented:
> > Thanks @komuw I have approved the PR
> 
> Thanks
> 
> > once approved by @mitjaziv and @imogenkraak
> 
> During todays stand-up, Mitja elected not to review the PR. Instead it will be Imogen who will do the review.
> 
> > will merge and release
> 
> Will this show up in the public docs? We would want to wait until the release of `Ara 1.18.0`, since this PR is made in the context of some monitoring changes that will only be available on Tyk cloud v1.18.0
> 
> cc @dcs3spp

@komuw When is Ara 1.18.0 scheduled for release? We can either:
- merge to main branch once approved and then release when Ara 1.18.0 scheduled or
- merge PR when 1.18.0 is scheduled (in which case we close the PR once approved and re-open ready for release of Ara 1.18.0)

imogenkraak commented on tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-usage.md:
```diff
@@ -0,0 +1,36 @@
+- **Throughput** - This is the total amount of data that has been transferred(ingress/egress) through a deployment. 
+- **Storage** - This is the total amount of data stored as [analytics by Tyk Dashboard]({{< ref "tyk-dashboard-analytics/" >}}). An example is *per request* statistics containing information about each request, like path or status.
+
+### How to check metrics?
```
Should remove the question mark here as it's not really a question

komuw replied:
fixed in https://github.com/TykTechnologies/tyk-docs/pull/4518/commits/b29886b4d147bd5cbd5ecb36970bc440a672949b

komuw commented on tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-how-it-works.md:
```diff
@@ -0,0 +1,40 @@
+---
+title: "What Metrics Are Monitored?"
```
@imogenkraak fixed in https://github.com/TykTechnologies/tyk-docs/pull/4518/commits/b29886b4d147bd5cbd5ecb36970bc440a672949b

imogenkraak commented on tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-how-it-works.md:
```diff
@@ -0,0 +1,40 @@
+
+{{< img src="/img/cloud/tyk-cloud-monitoring-priced-traffic.png" alt="Monitoring Traffic Pricing" >}}
+
+### Billed traffic
```
I'm not sure about this so just a suggestion, but it feels like `Billed traffic` and `Unbilled traffic` sections shouldn't be in the middle of `Throughput` and `Storage`. Could this belong as it's own `### Traffic billing` section, with `#### Billed traffic` and `####  Unbilled traffic` subsections?

komuw replied:
@imogenkraak It is under the throughput section because it is only for throughput where we have a distinction where some traffic is billed and others is not. This is not the case for storage.

imogenkraak replied:
maybe it should be a subsection under throughput? Or have the titles of billed traffic and unbilled be a bit smaller? But can also keep it like this, I'm just being pedantic, I will approve :)

dcs3spp replied:
Thanks @imogenkraak for raising the issue 👍 I have pushed changes to the branch so that titles for billed traffic and unbilled are smaller within the throughput section.

@komuw @imogenkraak if you are both ok with the change I will merge to the main branch ready for release

imogenkraak commented on tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-usage.md:
```diff
@@ -0,0 +1,36 @@
+---
+title: "How to view usage"
```
Could also just be `Viewing usage`

komuw replied:
Fixed in https://github.com/TykTechnologies/tyk-docs/pull/4518/commits/87f3eb33f22a1ee3c6029a238cf4046094307fa2

imogenkraak commented on tyk-docs/data/menu.yaml:
```diff
@@ -306,9 +306,17 @@ menu:
+              path: /tyk-cloud/environments--deployments/monitoring-how-it-works
+              category: Page
+              show: True
+            - title: "How to view usage"
```
Could also just be `Viewing usage`

komuw replied:
Fixed in https://github.com/TykTechnologies/tyk-docs/pull/4518/commits/87f3eb33f22a1ee3c6029a238cf4046094307fa2

imogenkraak commented on tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-usage.md:
```diff
@@ -0,0 +1,36 @@
+
+{{< img src="/img/cloud/tyk-cloud-monitoring-throughput.png" alt="Monitoring Throughput" >}}
+
+Enable *Storage* to display throughput metrics
```
fix to display `display storage metrics` instead of `throughput`

komuw replied:
Fixed in https://github.com/TykTechnologies/tyk-docs/pull/4518/commits/87f3eb33f22a1ee3c6029a238cf4046094307fa2 
Thanks.

komuw commented:
@dcs3spp 

> When is Ara 1.18.0 scheduled for release? 

Tentatively, in three weeks time.

> We can either:
> - merge to main branch once approved and then release when Ara 1.18.0 scheduled or
> - merge PR when 1.18.0 is scheduled (in which case we close the PR once approved and re-open ready for release of Ara 1.18.0)

I'm fine with either option, though I would prefer the first so that I don't have to deal with merge conflicts in the future. But I'll let you make the call.

imogenkraak commented on tyk-docs/content/tyk-cloud/environments-&-deployments/monitoring-usage.md:
```diff
@@ -0,0 +1,36 @@
+
+{{< img src="/img/cloud/tyk-cloud-monitoring-throughput.png" alt="Monitoring Throughput" >}}
+
+Enable *Storage* to display storage metrics
```
Tiny comment: need a dot at the end of this sentence to match the others.

komuw replied:
fixed https://github.com/TykTechnologies/tyk-docs/pull/4518/commits/8e9f3f871e5f4eec6a05072f2eea790050528a44

dcs3spp commented:
Thanks for approving @imogenkraak 👍 

@komuw If you can address the [tiny comment ](https://github.com/TykTechnologies/tyk-docs/pull/4518#discussion_r1587286681) found by @imogenkraak then I will merge to the main branch ready for release. When Ara 1.18.0 is released then if you could tag me I will then release to live docs.

komuw commented:
> Thanks for approving @imogenkraak 👍
> 
> @komuw If you can address the [tiny comment ](https://github.com/TykTechnologies/tyk-docs/pull/4518#discussion_r1587286681) found by @imogenkraak then I will merge to the main branch ready for release. When Ara 1.18.0 is released then if you could tag me I will then release to live docs.

Fixed in https://github.com/TykTechnologies/tyk-docs/commit/8e9f3f871e5f4eec6a05072f2eea790050528a44

dcs3spp commented:
/release to release-5.3

dcs3spp commented:
/release to release-5.2

dcs3spp commented:
/release to release-5.1
